x/oracle/keeper/native_token_test.go

```diff
@@ -0,0 +1,252 @@
+package keeper_test
+
+import (
+	"encoding/binary"
+	"strings"
+
+	sdkmath "cosmossdk.io/math"
+
+	"github.com/ExocoreNetwork/exocore/x/oracle/types"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/imroc/biu"
+)
+
+// workflow:
+// 1. Deposit. into staker_A
+//  1. stakerInfo {totalDeposit, price} - new
+//  2. stakerList - new
+//
+// 2. Msg. minus staker_A's amountOriginal
+//  1. stakerInfo_A {price-change} - update
+//  2. operatorInfo_A {price-change} - update
+//
+// 3. Deposit more. into staker_A
+//  1. stakerInfo {totalDeposit-change} -update
+//
+// 4. Msg. add staker_A's amountOriginal
+//  1. stakerInfo_A {price-change} - update
+//  2. operatorInfo_A {price-change} - update
+//
+// 5. withdraw from staker_A
+//  1. revmoed validatorIndex from stakerInfo
+//
+// 6. withdraw all from staker_A
+//  1. removed stakerInfo
+//  2. removed stakerList
+
+func (ks *KeeperSuite) TestNativeTokenLifeCycleOneStaker() {
+	operator := ks.Operators[0]
+	stakerStr := common.Address(operator.Bytes()).String()
+	chainID := "beaconchain"
+	// TODO: update after assets module got the definition
+	assetID := "0xe_" + chainID
+	validators := []string{"0xv1", "0xv2"}
+	// 1. deposit amount 100
+	// 100 is not a possible nubmer with one validator, it's ok to use this as a start and we'll check the number to be updated to a right number(uncer 32 with one validator)
+	amount100 := sdkmath.NewIntFromUint64(100)
+	amount32 := sdkmath.NewIntFromUint64(32)
+	ks.k.UpdateNativeTokenValidatorListForStaker(ks.ctx, chainID, stakerStr, validators[0], amount100)
+	// - 1.1 check stakerInfo
+	stakerInfo := ks.k.GetStakerInfo(ks.ctx, assetID, stakerStr)
```

Comment:
no case when GetStakerInfo is nil value return ?

Reply:
line129-130 is nil case: the staker has been removed and return value is empty struct( when nil)